test/SerialIOTest.cpp

```diff
@@ -543,9 +553,24 @@ close_iteration_interleaved_test( std::string file_ending )
 
 TEST_CASE( "close_iteration_interleaved_test", "[serial]" )
 {
+    bool const bp_prefer_adios1 =
+        ( auxiliary::getEnvString( "OPENPMD_BP_BACKEND", "NOT_SET" ) == "ADIOS1" );
+
     for( auto const & t : testedFileExtensions() )
     {
-        close_iteration_interleaved_test( t );
+        //! @FIXME ADIOS1 bugs with Iteration::close()
+        if( bp_prefer_adios1 )
+            continue;
+
+        close_iteration_interleaved_test( t, IterationEncoding::fileBased );
+        close_iteration_interleaved_test( t, IterationEncoding::groupBased );
+
+        // run this test for ADIOS2 & JSON only
+        if( t == "h5" )
+            continue;
+        if( t == "bp" && bp_prefer_adios1 )
+            continue;
+        close_iteration_interleaved_test( t, IterationEncoding::variableBased );
```

Comment:
Here we ideally should throw front-end errors that we can easily catch when `write.setIterationEncoding` is called for non-ADIOS2 backends.

Curiously, JSON just works (but is probably not doing something else than `groupBased`, from what I saw in the diff?).

Reply:
Quoting @franzpoeschel on slack: The basic idea was that this works for other backends as long as they only write one iteration:
https://github.com/openPMD/openPMD-api/blob/56c5353dd1354a100d941fc9f11a7f9c13c0c9ed/test/SerialIOTest.cpp#L3592

Here we write multiple.
Maybe we can guard somehow about this, not sure if/how we documented that `variableBased` works for only one iteration for other backends :)